expfmt/expfmt.go

```diff
@@ -70,6 +71,26 @@ const (
 	TypeOpenMetrics
 )
 
+// NewFormat generates a new Format from the type provided. Mostly used for
+// tests, most Formats should be generated as part of content negotiation in
+// encode.go.
+func NewFormat(t FormatType) Format {
+	switch t {
+	case TypeProtoCompact:
+		return fmtProtoCompact
+	case TypeProtoDelim:
+		return fmtProtoDelim
+	case TypeProtoText:
+		return fmtProtoText
+	case TypeTextPlain:
+		return fmtText
+	case TypeOpenMetrics:
```

Comment:
Is there a way to generate a fmtOpenMetrics_0_0_1?

Reply:
yeah, need the same here https://github.com/kumahq/kuma/pull/9462/files#diff-7f7235c97e7b47eb33457b82a7d6a9ae72052d0fe18249e47c58531a3dd7ef64R37

Reply:
yes, the new functions in expfmt.go can create these for you: https://github.com/prometheus/common/blob/main/expfmt/expfmt.go#L77

The story is, because the exposition formats are getting more complex, simple string comparison is no longer safe. There are too many equivalent permutations, and therefore things were moved to functions.  If the available NewFormat function is not sufficient for your needs, please @ me or file an issue and I can fix it up

Reply:
Right now TypeOpenMetrics only generates fmtOpenMetrics_1_0_0

If for some reason you need fmtOpenMetrics_0_0_1 there's no way to generate it

Reply:
I see -- yup I can prep a PR

Reply:
it looks like I should also increase the test coverage -- I'll get to this next week. sorry for the breakage in the meantime!

Reply:
@mrueg please let me know if this approach works for you

https://github.com/prometheus/common/pull/596